device/tt_cluster_descriptor.h

```diff
@@ -36,6 +36,7 @@ class tt_ClusterDescriptor {
 
   protected:
 
+  bool chips_have_coordinates;
```

Comment:
Or, we could fill chip_locations if "chips" is defined (which we do right now), but also change the check of chips_have_coordinates to !chip_locations.empty()